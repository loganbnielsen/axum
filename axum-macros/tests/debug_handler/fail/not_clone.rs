use axum_macros::debug_handler;
use axum::extract::Extension;

struct NonCloneType;

#[debug_handler]
async fn test_extension_non_clone(_: Extension<NonCloneType>) {}

fn main() {}
