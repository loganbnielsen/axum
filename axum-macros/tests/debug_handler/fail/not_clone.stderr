error[E0277]: the trait bound `NonCloneType: Clone` is not satisfied
 --> tests/debug_handler/fail/not_clone.rs:7:38
  |
7 | async fn test_extension_non_clone(_: Extension<NonCloneType>) {}
  |                                      ^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `NonCloneType`
  |
  = note: required for `Extension<NonCloneType>` to implement `Clone`
  = help: see issue #48214
  = help: add `#![feature(trivial_bounds)]` to the crate attributes to enable
help: consider annotating `NonCloneType` with `#[derive(Clone)]`
  |
4 + #[derive(Clone)]
5 | struct NonCloneType;
  |
